### PR TITLE
Streams sample

### PR DESCRIPTION
**Pull request by hananiel**
Samples for streams

### CODE REVIEW

File: Streams/Basic/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#See https://aka.ms/containerfastmode to understand how Visual Studio uses this Dockerfile to build your images for faster debugging.
+
+FROM mcr.microsoft.com/dotnet/core/runtime:3.1-buster-slim AS base
+WORKDIR /app
+
+FROM mcr.microsoft.com/dotnet/core/sdk:3.1-buster AS build
+WORKDIR /src
+COPY ["Basic.csproj", "Basic/"]
+RUN dotnet nuget add source https://pkgs.dev.azure.com/dotnet/Steeltoe/_packaging/dev/nuget/v3/index.json -n SteeltoeDe
```
Comment (TimHess) [inline Streams/Basic/Dockerfile:9]: Dev feed isn't needed for versions in use

File: Streams/CloudDataFlowToUpperProcessor/Program.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace CloudDataflowToUpperProcessor
+{
+    [EnableBinding(typeof(IProcessor))]
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+
+            var host = CreateStreamHostBuilder(args).Build(); // Alternatively use the default host builder
```
Comment (TimHess) [inline Streams/CloudDataFlowToUpperProcessor/Program.cs:23]:
```suggestion
            var host = CreateStreamHostBuilder(args).Build();
```

File: Streams/CloudDataFlowToUpperProcessor/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
..."vhost\": \"9cd41ebb-ff50-42b7-83f7-646291498fd8\" }, \"management\": { \"host\": \"192.168.8.36\", \"hosts\": [ \"192.168.8.36\" ], \"password\": \"HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w\", \"path\": \"/api/\", \"port\": 15672, \"ssl\": false, \"uri\": \"http://1c6e4609-7685-415a-8d6a-4b11645330f7:HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w@192.168.8.36:15672/api/\", \"uris\": [ \"http://1c6e4609-7685-415a-8d6a-4b11645330f7:HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w@192.168.8.36:15672/api/\" ], \"username\": \"1c6e4609-7685-415a-8d6a-4b11645330f7\" }, \"mqtt\": { \"host\": \"192.168.8.36\", \"hosts\": [ \"192.168.8.36\" ], \"password\": \"HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w\", \"port\": 1883, \"ssl\": false, \"uri\": \"mqtt://9cd41ebb-ff50-42b7-83f7-646291498fd8%3A1c6e4609-7685-415a-8d6a-4b11645330f7:HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w@192.168.8.36:1883\", \"uris\": [ \"mqtt://9cd41ebb-ff50-42b7-83f7-646291498fd8%3A1c6e4609-7685-415a-8d6a-4b11645330f7:HDv2c37a88vn_gS0Oyw7clwnXtfKxp9w@192.168.8.36:1883\" ], \"username\": \"9cd41ebb-ff50-42b7-83f7-646291498fd8:1c6e4609-7685-415a-8d6a-4b11645330f7\" } }, \"username\": \"1c6e4609-7685-415a-8d6a-4b11645330f7\" }, \"syslog_drain_url\": null, \"volume_mounts\": [ ]}]}",
```
Comment (TimHess) [inline Streams/CloudDataFlowToUpperProcessor/Properties/launchSettings.json:17]: VCAPs again

File: Streams/CloudDataFlowToUpperProcessor/Controllers/WeatherForecastController.cs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            return forecast;
+        }
+        [HttpGet("Config")]
+        public BindingServiceOptions GetConfig()
+        {
+            return this.bindingsOptions.CurrentValue ;
+        }
+
+        [HttpGet("RabbitOptions")]
+        public string GetconnectionFactory()
```
Comment (TimHess) [inline Streams/CloudDataFlowToUpperProcessor/Controllers/WeatherForecastController.cs:71]: If this is staying, should probably add a comment that it is not a good practice

File: Streams/CloudDataFlowToUpperProcessor/CloudDataflowToUpperProcessor.csproj [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="5.0.1" NoWarn="NU1605" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.OpenIdConnect" Version="5.0.1" NoWarn="NU1605" />
+    <PackageReference Include="Microsoft.VisualStudio.Azure.Containers.Tools.Targets" Version="1.10.9" />
+    <PackageReference Include="Swashbuckle.AspNetCore" Version="5.6.3" />
+    <PackageReference Include="Steeltoe.Common" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Common.Hosting" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Connector.CloudFoundry" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Connector.ConnectorBase" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Extensions.Configuration.CloudFoundryCore" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Management.CloudFoundryCore" Version="3.1.0-rc1" />
```
Comment (TimHess) [inline Streams/CloudDataFlowToUpperProcessor/CloudDataflowToUpperProcessor.csproj:21]:
```suggestion
    <PackageReference Include="Steeltoe.Management.EndpointCore" Version="3.1.0-rc1" />
```

File: Stream/Basic/Program.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+        [StreamListener(IProcessor.INPUT)]
+        [SendTo(IProcessor.OUTPUT)]
+        public string Handle(string inputVal)
+        {
+            return inputVal.ToUpper();
+        }
+
+    }
+
```
Comment (TimHess) [inline Stream/Basic/Program.cs:39]:
```suggestion
            await StreamHost.CreateDefaultBuilder<Program>(args)
              .ConfigureServices((context, services) =>
              {
                  services.AddLogging(builder =>
                  {
                      builder.AddDebug();
                      builder.AddConsole();
                  });
              }).RunConsoleAsync();
        }

        [StreamListener(IProcessor.INPUT)]
        [SendTo(IProcessor.OUTPUT)]
        public string Handle(string inputVal)
        {
            return inputVal.ToUpper();
        }
    }
```
whitespace

File: Stream/CloudDataFlowLogSink/Startup.cs [hunk 7]
```diff
@@ -0,0 +1,56 @@
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+//using Microsoft.OpenApi.Models;
```
Comment (TimHess) [inline Stream/CloudDataFlowLogSink/Startup.cs:8]:
```suggestion
```

In the interest of keeping samples concise... Are all the other namespaces used?

File: Stream/CloudDataFlowLogSink/Startup.cs [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
```
Comment (TimHess) [inline Stream/CloudDataFlowLogSink/Startup.cs:47]:
```suggestion
```
Auth doesn't appear to matter for this sample

File: Stream/CloudDataFlowToUpperProcessor/Controllers/WeatherForecastController.cs [hunk 9]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using RabbitMQ.Client;
+using Steeltoe.Common.Util;
```
Comment (TimHess) [inline Stream/CloudDataFlowToUpperProcessor/Controllers/WeatherForecastController.cs:10]: Most of these namespaces aren't needed

File: Stream/CloudDataFlowToUpperProcessor/Startup.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+                c.SwaggerDoc("v1", new OpenApiInfo { Title = "DataflowToUpper", Version = "v1" });
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+
+
+
```
Comment (TimHess) [inline Stream/CloudDataFlowToUpperProcessor/Startup.cs:38]:
```suggestion
```

File: Stream/LoggingConsumerApplication/Nuget.config [hunk 11]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (TimHess) [inline Stream/LoggingConsumerApplication/Nuget.config:1]: delete file?

File: Stream/CloudDataFlowToUpperProcessor/CloudDataflowToUpperProcessor.csproj [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+    <PackageReference Include="Steeltoe.Common.Hosting" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Connector.CloudFoundry" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Connector.ConnectorBase" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Extensions.Configuration.CloudFoundryCore" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Management.EndpointCore" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Stream.Binder.RabbitMQ" Version="3.1.0-rc1" />
+    <PackageReference Include="Steeltoe.Stream.StreamBase" Version="3.1.0-rc1" />
+  </ItemGroup>
+
+
```
[the remark this answers is not in the document]
Comment (hananiel) [reply Stream/CloudDataFlowToUpperProcessor/CloudDataflowToUpperProcessor.csproj]: Everything except common is needed.

File: Stream/EFCore/Readme.md [hunk 13]
```diff
@@ -0,0 +1 @@
+﻿ docker run --env MYSQL_ROOT_PASSWORD=my-secret-pw --publish 3306:3306 steeltoeoss/mysql
```
[the remark this answers is not in the document]
Comment (hananiel) [reply Stream/EFCore/Readme.md]: Removed the anemic readmes for now - will come back to add more detail later.

File: Stream/PolledConsumers/ReadMe.md [hunk 14]
```diff
@@ -0,0 +1,3 @@
+﻿Find Destin.anonymous queue //TODO: named
```
[the remark this answers is not in the document]
Comment (hananiel) [reply Stream/PolledConsumers/ReadMe.md]: Deleting these readmes for now.

**Review by TimHess — APPROVED** [review]

**Pull request merged**